### DISCUSSION TITLE
Unicode problem -> invalid input syntax for type json

### DISCUSSION THREAD

**Discussion by hilmanski** [supabase/supabase#7499] [Questions]
Hi I want to insert his data on supabase
```
{
  thread_id: '907795951207456788',
  user: { username: 'Fedor', user_id: '123' },
  title: 'a text +
    '\n' +
    'Steam page is live! 🙂 \n' +
    '\n' +
    '🎉 🎉 ',
  slug: 'text-slug,
  body: 'text-slug\n' +
    '\n' +
    'Steam page is live! 🙂 \n' +
    '\n' +
    '🎉 🎉 \ud83c',
  type: 'DEFAULT',
  reference: null,
  channel_id: '123',
  guild_id: '123',
  created_at: '2021-11-10T00:57:00.801Z'
}
```

Error log shows
```
{
  message: 'invalid input syntax for type json',
  code: '22P02',
  details: 'Unicode low surrogate must follow a high surrogate.',
  hint: null
}
```

When I play with the "\ud83c" text. 
- It can be inserted when I remove this on body
- It's a problem when I have this whether in body or in title (varchar)

**Comment by steve-chavez** [comment]
Can you convert that json string to use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)? It's easier to copy/paste that way.

wrt to the issue, if you can do this with pure SQL:

```sql
select '"My long string 🙂 with emojis 🎉 \ud83c"'::json;
```

Then the insertion should succeed. From what I've tested, the above works fine.

**Comment by hilmanski** [comment]
Thanks @steve-chavez . 
I guess this error has nothing to do with supabase as well.
Will close it for now

**Reply by steve-chavez** [reply]
> {
  message: 'invalid input syntax for type json',
  code: '22P02',
  details: 'Unicode low surrogate must follow a high surrogate.',
  hint: null
}

Just to be clear, the above error does come from PostgreSQL.

**Reply by hilmanski** [reply]
yap, correct.
